README.md

```diff
@@ -1,146 +1,195 @@
-            @Override
-            public void onSuccess(AuthenticationResult authenticationResult) {
-                /* Successfully got a token, use it to call a protected resource */
+```java
 
-                String accessToken = authenticationResult.getAccessToken();
-            }
-            @Override
-            public void onError(MsalException exception) {
-                /* Failed to acquireToken */
-
-                if (exception instanceof MsalClientException) {
-                    /* Exception inside MSAL, more info inside MsalError.java */
-                } else if (exception instanceof MsalServiceException) {
-                    /* Exception when communicating with the STS, likely config issue */
-                }
-            }
-            @Override
-            public void onCancel() {
-                /* User canceled the authentication */
+mMultipleAccountApp.acquireToken(this, SCOPES, getAuthInteractiveCallback());
+
+private AuthenticationCallback getAuthInteractiveCallback() {
+    return new AuthenticationCallback() {
+        @Override
+        public void onSuccess(AuthenticationResult authenticationResult) {
+            /* Successfully got a token, use it to call a protected resource */
+            String accessToken = authenticationResult.getAccessToken();
+            // Record account used to acquire token
+            mFirstAccount = authenticationResult.getAccount();
+        }
+        @Override
+        public void onError(MsalException exception) {
+            if (exception instanceof MsalClientException) {
+                //And exception from the client (MSAL)
+            } else if (exception instanceof MsalServiceException) {
+                //An exception from the server
             }
-        };
-    }
+        }
+        @Override
+        public void onCancel() {
+            /* User canceled the authentication */
+        }
+    };
+}
 ```
 
-### Step 4: Use the token!
+3. Acquire a token silently
 
-The access token can now be used to [Call an API](https://github.com/AzureAD/microsoft-authentication-library-for-android/wiki/Calling-an-API)).
+```java
 
-## Running Tests
+/*
+    Before getting a token silently for the account used to previously acquire a token interactively.  We recommend that you verify that the account is still present in the local cache or on the device in case of brokered auth
 
-** NEED
+    Let's use the synchronous methods here which can only be invoked from a Worker thread
+*/
+
+//On a worker thread
+IAccount account = mMultipleAccountApp.getAccount(mFirstAccount.getId());
+
+if(account != null){
+    //Now that we know the account is still present in the local cache or not the device (broker authentication)
+
+    //Request token silently
+    String[] newScopes = {"Calendars.Read"}
+    
+    //Use default authority to request token from pass null
+    IAuthenticationResult result = mMultipleAccountApp.acquireTokenSilent(newScopes, account, null);
```

Comment:
we are validating authority to be `NonNull` on silent call

Reply:
Yerp.

https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/dev/msal/src/main/java/com/microsoft/identity/client/MultipleAccountPublicClientApplication.java#L91